packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -503,6 +506,20 @@ function createAssumeRolePolicy(principal: IPrincipal, externalIds: string[]) {
   return actualDoc;
 }
 
+function validateRolePath(path?: string) {
+  if (path === undefined || Token.isUnresolved(path)) {
+    return;
+  }
+
+  const validRolePath = /^(\/|\/[\u0021-\u007F]+\/)$/;
+
+  if (path.length == 0 || path.length > 512) {
+    throw new Error(`role path is set to ${path}, but the length must be >= 1 and <= 512.`);
```

Comment:
```suggestion
    throw new Error(`Role path must be between 1 and 512 characters. The provided role path is ${path.length} characters.`);
```

Reply:
Fixed in [8d9afeb](https://github.com/aws/aws-cdk/pull/16165/commits/8d9afeb17d0128c50ba98a5fa6645993dd1415ce)